spec/models/swift_storage_provider_spec.rb

```diff
@@ -308,5 +308,46 @@
       is_expected.to validate_presence_of :chunk_max_number
       is_expected.to validate_presence_of :chunk_max_size_bytes
     end
+
+    context 'is_default' do
+      let(:new_default_storage_provider) { FactoryBot.build(:swift_storage_provider, :default) }
+      let(:new_not_default_storage_provider) { FactoryBot.build(:swift_storage_provider) }
+      it 'should allow only one default storage_provider' do
+        expect(subject.is_default?).to be_truthy
+        expect(new_default_storage_provider).not_to be_valid
+        expect(new_not_default_storage_provider).to be_valid
+        subject.update(is_default: false)
+        expect(new_default_storage_provider).to be_valid
+        expect(new_not_default_storage_provider).to be_valid
+      end
+    end
+
+    context 'is_deprecated' do
+      let(:not_default_storage_provider) { FactoryBot.build(:swift_storage_provider) }
+      it {
+        is_expected.to be_valid
+        expect(not_default_storage_provider).to be_valid
+
+        subject.is_deprecated = true
+        is_expected.not_to be_valid
+
+        not_default_storage_provider.is_deprecated = true
+        expect(not_default_storage_provider).to be_valid
+      }
+    end
+  end
+
+  describe '.default' do
+    let(:default_storage_provider) { FactoryBot.create(:swift_storage_provider, :default) }
+    let(:not_default_storage_provider) { FactoryBot.create(:swift_storage_provider) }
+    subject { StorageProvider.default }
+    it { expect(described_class).to respond_to(:default) }
+    it {
+      expect(default_storage_provider).to be_persisted
```

Comment:
What happens when `default_storage_provider` isn't persisted?

Reply:
it returns nil and fails the test